app/controllers/pieces_controller.rb

```diff
@@ -30,14 +35,20 @@ def update
   end
 
   def create
-    @piece = Piece.new(piece_params)
-    @piece.user = current_user
-    if @piece.save
-      flash[:notice] = "Piece added successfully"
-      redirect_to piece_path(@piece)
-    else
-      flash[:errors] = @piece.errors.full_messages.join(". ")
+    if !user_signed_in?
+      flash.now[:notice] = "Please sign in to add a piece of music"
+      @piece = Piece.new
       render :new
+    else
```

Comment:
Revert this action to how it was, if a use is not authenticated then saving the piece will fail on the user validation of the piece, and hence they'll see the appropriate error message.